src/ml/neural_net/weight_init.cpp

```diff
@@ -35,6 +35,13 @@ void xavier_weight_initializer::operator()(float* first_weight,
   }
 }
 
+void ones_weight_initializer::operator()(float* first_weight,
```

Comment:
Could these be passed as const references?

Reply:
Probably just the last one.  We use the first one as an iterator of sorts and increment the pointer. I'd have to let @nickjong speak to this. He probably has a better context to this.

Reply:
Oh, right. So maybe neither of them can be references, but I think they should be const.

Reply:
So I was referring to the const. Since we use the first pointer as an iterator, not sure it can be a const unless we want to do a `const_cast` later. I remember that being inadvisable. But again I'd have to deffer that to nick. Basically with the current implementation making them const means that the program wouldn't compile.

Reply:
These types cannot be `const`, since they specify a range of mutable memory that we want this function to initialize. We write to each `float` in the range in line 41. That line would generate a compiler error if `w` had type `const float *`